src/main/resources/com/google/cloud/tools/libraries/libraries.json

```diff
@@ -526,7 +526,43 @@
       "mavenCoordinates": {
          "groupId": "com.google.cloud",
          "artifactId": "google-cloud-dialogflow",
-         "version": "0.48.0-alpha"
+         "version": "0.50.0-alpha"
+       }
+      }
+    ]
+  },
+{
+    "name": "IoT Core",
+    "id": "iotcore",
+    "serviceName": "cloudiot.googleapis.com",
+    "serviceRoles": [
+      "roles/cloudiot.admin",
+      "roles/owner",
```

Comment:
this role looks suspiciously out of place for IoT. like a general owner role?

Reply:
same for `roles/editor`, `roles/viewer`

Reply:
the docs you linked to say you can use `roles/cloudiot.editor` instead of `roles\editor`. similar for the others.

Reply:
Is this supposed to be an exhaustive list? I simply added what was on that page.

Reply:
right, but I'm just saying we should used the cloudiot scoped roles and not the general ones like `roles/editor` since those probably grant much broader permissions.

Reply:
from the first link:

"Note that the roles roles/owner, roles/editor, and roles/viewer include permissions for other Google Cloud Platform services as well. "

I'm not sure why they even listed those

Reply:
What's the purpose of including these roles in the metadata? (Eclipse doesn't use them.)

Reply:
IntelliJ has the feature, during client library addition, to create new service accounts and assign specific roles to them.

For example, if the user selects to add the datastore client library, we present the option to create a new service account and display the associated datastore roles (with the option to pick and choose how many or how little to apply)

Reply:
So the question becomes, should IntelliJ offer these roles or not? It seems like we might want them on a service account, or are they deprecated somehow? How do they apply to other APIs where we don't currently list these roles? 

If they shouldn't be used, maybe we should file a bug to correct the  https://cloud.google.com/iot/docs/how-tos/iam page.

Reply:
lgtm now. I think filing a bug on that page is a good idea. It doesn't seem appropriate to have those roles there. I'll do taht.